server/api/src/rsGenQuery.cpp

```diff
@@ -214,7 +212,31 @@ namespace {
         }
 
         return fmt::format("IN ({})", fmt::join(leaf_ids, ","));
-    } // get_resc_id_cond_for_hier_cond
+    } // translate_single_data_resc_hier_condition_to_resc_id
+
+    auto translate_data_resc_hier_where_clause_to_resc_id(const std::string& _condition) -> std::string
+    {
+        std::string condition_str;
+
+        // For each || operator found, take the clause immediately preceding and translate it from the DATA_RESC_HIER
+        // condition to the equivalent RESC_ID condition.
+        auto previous_or_position = std::string_view::size_type{};
+        for (auto or_position = _condition.find("||", previous_or_position); std::string_view::npos != or_position;
+             or_position = _condition.find("||", previous_or_position))
+        {
+            condition_str += translate_single_data_resc_hier_condition_to_resc_id(
+                                 _condition.substr(previous_or_position, or_position)) +
+                             " || ";
+
+            // Advance the previous position tracker to the current || position, plus 2 so that the next clause does not
+            // include the || itself.
+            previous_or_position = or_position + 2;
+        }
+
+        // There should be one clause found after the last || operator has been found which will need to be translated.
```

Comment:
Is this really true?

Given something like `|| like A || like B || like C`, `or_position` will visit each `||` before returning `std::string_view::npos`.

What am I missing here?

Reply:
The comment on 221-222 reads...
> For each || operator found, take the clause immediately preceding and translate it from the DATA_RESC_HIER condition to the equivalent RESC_ID condition.

So, the loop captures the strings *before* the `||` and translates it to the equivalent RESC_ID condition. We need this part at the end to translate the *final* right-hand side operand of the binary `||` operator because the loop is translating conditions that *precede* the `||` operator.

To tweak the example a bit, if we had:
`DATA_RESC_HIER like 'A' || like 'B' || like 'C'`

The `like 'C'` condition is what is being translated here. The loop handles `like 'A'` and `like 'B'`.

Happy to try other approaches as well

Reply:
If it helps, I seem to have been misusing the word "clause" in this and a couple of other places. I will update the comments to appropriately refer to these pieces as "conditions" (where the set of conditions compose the "clause").

Reply:
Sure thing.